File: app/src/main/java/com/weathercock/cleanarchitecture/data/api/WeatherApi.java
package com.weathercock.cleanarchitecture.data.api;

import com.weathercock.cleanarchitecture.data.entity.WeatherEntity;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Weather API from livedoor
 * http://weather.livedoor.com/weather_hacks/webservice
 * <p>
 * Created by tamura_k on 2016/06/03.
 */
public interface WeatherApi {

    @GET("json/v1")
    Observable<WeatherEntity> get(@Query("city") int cityId);

}

File: app/src/main/java/com/kazakago/cleanarchitecture/web/api/WeatherApi.kt
package com.kazakago.cleanarchitecture.web.api

import com.kazakago.cleanarchitecture.web.entity.weather.WeatherApiEntity
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherApi {

    @GET("json/v1")
    operator fun get(@Query("city") cityId: String): Single<WeatherApiEntity>

}
